# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/parted/hotfix-hfs.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- parted-3.6/libparted/fs/r/hfs/hfs.c.vanilla	2023-07-18 20:15:46.408681000 +0500
+++ parted-3.6/libparted/fs/r/hfs/hfs.c	2023-07-18 20:17:32.592681000 +0500
@@ -193,7 +193,8 @@
                 return 0;
 #endif
 
-	PED_ASSERT ((hgee = hfs_get_empty_end(fs)) != 0);
+	hgee = hfs_get_empty_end(fs);
+	PED_ASSERT (hgee != 0);
 
 	if (ped_geometry_test_equal(fs->geom, geom))
 		return 1;
